packages/ra-core/src/form/useChoices.tsx

```diff
@@ -62,7 +62,7 @@ export const useChoices = ({
                     : get(choice, optionText);
 
             return translateChoice
-                ? translate(choiceName, { _: choiceName })
+                ? translate(String(choiceName), { _: choiceName })
```

Comment:
I don't think that's the right fix. If the `optionText` is a valid react element, we shouldn't try to rtanslate it.

Reply:
Right, I forgot the use case you are talking about is fixed in another PR